GPflowOpt/optim.py

```diff
@@ -109,47 +109,27 @@ def silent(self):
         sys.stdout = save_stdout
 
 
-class CandidateOptimizer(Optimizer):
+class MCOptimizer(Optimizer):
     """
-    Optimization of an objective function by evaluating a set of pre-defined candidate points.
+    Optimization of an objective function by evaluating a set of random points.
 
-    Returns the point with minimal objective value.
+    Note: each call to optimize, a different set of random points is evaluated.
 
     For compatibility with the StagedOptimizer, the candidate points are concatenated with
     the initial points and evaluated.
```

Comment:
Actually this makes no sense at all. To pass on an optimum in the stagedoptimizer, this means it should be re-evaluated by the next optimizer? I guess its just inherent if you use a candidate/mcoptimizer there is no starting point. StagedOptimizer could simply return the best point it encountered over the entire pipeline.